source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,102 @@ Insert or Update in a Single Operation
-``upsert`` option available in the following methods:
+Overview
+--------
+
+You can insert or update documents by passing a query filter to the 
+:java-docs:`updateOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateOne(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`, 
+:java-docs:`updateMany() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateMany(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`
+or :java-docs:`replaceOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#replaceOne(org.bson.conversions.Bson,TDocument,com.mongodb.client.model.ReplaceOptions)>` methods
+
+Each method returns an ``AcknowledgedUpdateResult`` that tells us:
+
+- The number of documents that matched the query filter
+- The number of documents that got modified 
+- The ``_id`` of the document that got upserted 
+
+These methods update documents that match your query filter, but
+inserts a document if the there are no matches when you set the upsert
+to true.  
+
+In the case there are multiple matches in your query filter for
+``updateOne()`` or ``replaceOne()``, the method will update the first
+occurrence of a match in the collection.
+
+Set an upsert:
+~~~~~~~~~~~~~~
+
+Use :java-docs:`UpdateOptions </apidocs/mongodb-driver-core/com/mongodb/client/model/UpdateOptions.html>`
+as a third parameter to the ``updateOne()`` and ``updateMany()``
+methods.
+
+Use :java-docs:`ReplaceOptions </apidocs/mongodb-driver-core/com/mongodb/client/model/ReplaceOptions.html>`
+as a third parameter to the ``replaceOne()`` method.
+
+Example
+~~~~~~~
+
+The following example is about a paint store that sells eight different
+colors of paint. The store had their annual online sale resulting in the
+following documents in their ``paint_inventory`` collection: 
+
+.. code-block:: json
+
+    { "_id": { "$oid": "606b4cfbcd83be7518b958da" }, "color": "red", "qty": 5 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958db" }, "color": "purple", "qty": 8 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958dc" }, "color": "blue", "qty": 0 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958dd" }, "color": "white", "qty": 0 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958de" }, "color": "yellow", "qty": 6 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958df" }, "color": "pink", "qty": 0 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958e0" }, "color": "green", "qty": 0 }
+    { "_id": { "$oid": "606b4cfbcd83be7518b958e1" }, "color": "black", "qty": 8 }
+
+The store is restocking their items.  The first shipment that arrived
+are ten cans of orange paint and we need to update this in the collection.
+
+To update orange, query the ``paint_inventory`` collection where the
+``color`` is ``"orange"``, set the update for ``qty`` to ``10``, set
+``UpdateOptions.upsert()`` to ``true`` and pass the query to the
+``updateOne()`` method: 
+
```

Comment:
Looking at this again, I noticed there's some information here that doesn't really add anything. To summarize:

1. The store received a shipment of inventory.
2. The store uses upsert operations so that new inventory is added, or existing inventory is updated in a single operation.

If that's the case, maybe something along the lines of
The store received a fresh shipment and needs to update their inventory. By using an upsert, existing documents will be updated and new documents will be created in a single operation, simplifying the logic in their inventory control application.

The first item in the shipment is ten cans of orange paint. To update the inventory, or insert, query the ``paint_inventory`` collection for the appropriate color, specify an ``increment`` operation of 10 for the ``qty`` field in the update, and specify ``true`` to ``UpdateOptions.upsert()``.

Something along the lines of that. I'm unsure the value of stating the ``updateOne`` method since the example shows the method used.